File: Lab_02/Pizza.java
class Pizza {
  private int numToppings;
  private int slicesLeft = 8;

  public Pizza() {
    this.numToppings = 1; //cheese duh
  }

  public Pizza(int numToppings) {
    this.numToppings = numToppings;
  }

  public boolean eatSlice() {
    if (slicesLeft > 0) {
      --slicesLeft;
      return true;
    }
    return false;
  }

  public static void main (String[] args) {
    Pizza p = new Pizza(); // pizza with just cheese :(
    Pizza p2 = new Pizza(4); // pizza with four tops
    p.eatSlice();
    p2.eatSlice();
    System.out.println("Yum!");
  }

}

File: Lab_02/header-example2/pizza.cpp
#include "pizza.h"

int main() {
	Pizza p;
	std::cout << p.str() << std::endl;
	Pizza p2(234);
	p2.eat_slice();
	std::cout << p2.str() << std::endl;
}

void printHello() {
	std::cout << "hello" << std::endl;
}

Pizza::Pizza() {
	slicesLeft = 8;
	numToppings = 1;
}

Pizza::Pizza(int numToppings) : slicesLeft(8), numToppings(numToppings){}

bool Pizza::eat_slice() {
	if (slicesLeft > 0) {
		--slicesLeft;
		return true;
	}
	return false;
}

std::string Pizza::str() {
	return "Pizza Status=slicesLeft: " + std::to_string(slicesLeft) +
					" numToppings: " + std::to_string(numToppings);
}
